client/functions.lua

```diff
@@ -111,14 +112,86 @@ function QBCore.Functions.GetBlip(id)
     return blip
 end
 
-function QBCore.Functions.RequestAnimDict(animDict)
-	if not HasAnimDictLoaded(animDict) then
-		RequestAnimDict(animDict)
+function QBCore.Functions.LoadAnimDict(animDict)
+    if not HasAnimDictLoaded(animDict) then
+        RequestAnimDict(animDict)
 
-		while not HasAnimDictLoaded(animDict) do
-			Wait(4)
-		end
-	end
+        local timeout = false
+        SetTimeout(60000, function () timeout = false end)
+		
+        while not HasAnimDictLoaded(animDict) and not timeout do
+            Wait(0)
+        end
+    end
+end
+
+function QBCore.Functions.Animation:new(iPed, iType, iText, iDuration, iDict, iAnim, iFlag)
```

Comment:
This will break compatibility with other ScRT (javascript and c#) so maybe we shouldn't use metatables and this syntax for functions. Not to mention this doesn't fit with the style and conventions qbcore is already using